### PR TITLE
Improve logging and debugging in context of logs format

### PR DESCRIPTION
**Pull request by MalloZup**
# description

This PR:

1)  add `log_ok` and `log_error` helper function and adapt them in script. 

2) Print during execution only salt `error` and a resume of state at the end. 

3) create a `phase` logger a first v1 of logger, with colors which says the status of phase  with timestamp, hostname and msg

4) add back the colored output

We do have also a `debug` file for the full verbose mode in case needed .

# Screenshot:

image

image

### CODE REVIEW

**Comment by MalloZup** [comment]
I could do also https://github.com/SUSE/ha-sap-terraform-deployments/issues/528 on same PR but the `post-handling` script need to be done on other PRs but it is on the same context

File: salt/provision.sh [hunk 1]
```diff
@@ -16,6 +16,18 @@ get_grain () {
...
+  TIMESTAMP=`date -u`
+  GREEN='\033[0;32m'
+  RED='\033[0;31m'
+  NC='\033[0m' # No Color
+  printf "${GREEN}$TIMESTAMP::$NODE [INFO]  $1 ${NC}\n"
+}
+
+
+
+
```
Comment (stefanotorresi) [inline salt/provision.sh:30]: I think we can get rid of all these empty new lines :)

**Review by stefanotorresi — APPROVED** [review]
This looks like a very much needed change! GJ!

Only a those empty newlines to remove.

**Comment by MalloZup** [comment]
@stefanotorresi  thx for review. I want to push during the day more  to screenshot and some other minimal changes so will ask you re-review. and remove the wip

File: salt/provision.sh (hunk 1 above)
Comment (MalloZup) [reply salt/provision.sh:30]: yop

**Review by stefanotorresi — APPROVED** [review]
GLORIOUS! :grin: :star_struck: :grin:

**Comment by MalloZup** [comment]
thx @stefanotorresi .. :pray:  

@Simranpal @yandroskaos if from your perspective is ok? let me know your thoughts since it is a global change is always good to have different povs

**Review by yandroskaos — APPROVED** [review]
LGTM

**Pull request merged**